pkg/pool-hooks/contracts/MevHook.sol

```diff
@@ -223,4 +226,55 @@ contract MevHook is BaseHooks, SingletonAuthentication, VaultGuard, IMevHook {
 
         emit PoolMevTaxThresholdSet(pool, newPoolMevTaxThreshold);
     }
+
+    /// @inheritdoc IHooks
+    function onBeforeAddLiquidity(
+        address,
+        address pool,
+        AddLiquidityKind kind,
+        uint256[] memory,
+        uint256,
+        uint256[] memory,
+        bytes memory
+    ) public view override returns (bool success) {
+        if (_mevTaxEnabled == false) {
+            return true;
+        }
+
+        uint256 priorityGasPrice = _getPriorityGasPrice();
+
+        // Unbalanced adds are blocked for priority gas prices above the threshold.
+        if (kind != AddLiquidityKind.PROPORTIONAL && priorityGasPrice > _poolMevTaxThresholds[pool]) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        return kind == AddLiquidityKind.PROPORTIONAL || priorityGasPrice <= _poolMevTaxThresholds[pool];
```
Is this more readable? "Allow if proportional add or the price is under threshold"

Reply:
Also, consider moving this up to the other IHooks functions (e.g., after the dynamic swap fee hook), so that they're all together, and not after admin functions.